packages/ui/src/working-groups/hooks/useOpenings.ts

```diff
@@ -1,17 +1,34 @@
+import { useMemo } from 'react'
+
 import { useGetWorkingGroupOpeningsQuery } from '../queries'
-import { asWorkingGroupOpening } from '../types'
+import { asWorkingGroupOpening, WorkingGroupOpening } from '../types'
 
 interface UseOpeningsParams {
   groupId?: string
+  type?: 'open' | 'past'
 }
 
-export const useOpenings = ({ groupId }: UseOpeningsParams = {}) => {
+export const useOpenings = ({ groupId, type }: UseOpeningsParams = {}) => {
   const { loading, data } = useGetWorkingGroupOpeningsQuery({ variables: { group_eq: groupId } })
 
+  const typeFilter: (opening: WorkingGroupOpening) => boolean = useMemo(() => {
+    switch (type) {
+      case 'open':
+        return (opening) => isOpeningOpen(opening)
+      case 'past':
+        return (opening) => !isOpeningOpen(opening)
+      default:
+        return () => true
+    }
+  }, [type])
+
   const groups = data?.workingGroupOpenings ?? []
+  const openings = useMemo(() => groups.map(asWorkingGroupOpening).filter(typeFilter), [loading, data])
```

Comment:
After the above, the `typeFIlter` can be moved to the `useMemo()` so it will be invoked only after the data changes.

```suggestion
  const openings = useMemo(() => groups.map(asWorkingGroupOpening).filter(getTypeFiler(type)), [loading, data])
```